src/server/cluster/outgoing_slot_migration.cc

```diff
@@ -240,6 +248,9 @@ void OutgoingMigration::SyncFb() {
 }
 
 bool OutgoingMigration::FinalizeMigration(long attempt) {
+  OnAllShards([this](auto& migration) { server_family_->CancelBlockingOnThread(); });
+  Transaction::Guard tg{tx_.get()};
```

Comment:
why did you change this to use Transaction::Guard? this change means that we will not be able to run f.e the config update command until we finalize the migration

Reply:
We can't finalize a migration in-between a command running, we have to use a global transaction

> means that we will not be able to run f.e the config update command until we finalize the migration

It should be just enqueued in the transaction queue, so it will run immediately after

Reply:
Actually, client pause here is pretty safe from a tranasctional point 🤔 will revert back to it